crates/forge/tests/cli/zk/verify_bytecode.rs

```diff
@@ -0,0 +1,184 @@
+            hash_type: Some(BytecodeHash::None),
+            zksolc: Some(SolcReq::Version(Version::new(1, 5, 11))),
+            ..Default::default()
+        },
+        ..Default::default()
+    };
+
+    prj.add_source("Counter.sol", COUNTER_SOURCE).unwrap();
+
+    prj.write_config(config);
+
+    let args = vec![
+        "verify-bytecode",
+        COUNTER_ADDRESS,
+        "Counter",
+        "--rpc-url",
+        "https://sepolia.era.zksync.dev",
+        "--zksync",
+        "--verifier-url",
+        "https://block-explorer-api.sepolia.zksync.dev/api",
+    ];
+
+    let output = cmd.forge_fuse().args(args).assert_success().get_output().stdout_lossy();
+    assert!(output.contains("Runtime code matched with status partial"));
+});
+
+forgetest_async!(zk_verify_bytecode_ipfs, |prj, cmd| {
+    let config = Config {
+        evm_version: EvmVersion::London,
+        solc: Some(SolcReq::Version(Version::new(0, 8, 28))),
+        zksync: foundry_config::zksync::ZkSyncConfig {
+            hash_type: Some(BytecodeHash::Ipfs),
+            zksolc: Some(SolcReq::Version(Version::new(1, 5, 11))),
+            ..Default::default()
+        },
+        ..Default::default()
+    };
+
+    prj.add_source("Counter.sol", COUNTER_SOURCE).unwrap();
+
+    prj.write_config(config);
+
+    let args = vec![
+        "verify-bytecode",
+        COUNTER_ADDRESS,
+        "Counter",
+        "--rpc-url",
+        "https://sepolia.era.zksync.dev",
+        "--zksync",
+        "--verifier-url",
+        "https://block-explorer-api.sepolia.zksync.dev/api",
+    ];
+
+    let output = cmd.forge_fuse().args(args).assert_success().get_output().stdout_lossy();
+    assert!(output.contains("Runtime code matched with status partial"));
+});
+
+forgetest_async!(zk_verify_bytecode_error_diff, |prj, cmd| {
+    let config = Config {
+        evm_version: EvmVersion::Shanghai,
+        solc: Some(SolcReq::Version(Version::new(0, 8, 26))),
+        zksync: foundry_config::zksync::ZkSyncConfig {
+            zksolc: Some(SolcReq::Version(Version::new(1, 5, 10))),
+            optimizer: true,
+            optimizer_mode: '1',
+            ..Default::default()
+        },
+        ..Default::default()
+    };
+
+    prj.add_source("Counter.sol", COUNTER_SOURCE).unwrap();
+
+    prj.write_config(config);
+
+    let args = vec![
+        "verify-bytecode",
+        COUNTER_ADDRESS,
+        "Counter",
+        "--rpc-url",
+        "https://sepolia.era.zksync.dev",
+        "--zksync",
+        "--verifier-url",
+        "https://block-explorer-api.sepolia.zksync.dev/api",
+    ];
+
+    cmd.forge_fuse().args(args).assert_success().stderr_eq(
+        r#"Error: Runtime code did not match - this may be due to varying compiler settings
+EVM version mismatch: local=shanghai, onchain=london
+Optimizer mode mismatch: local=1, onchain=3
+"#,
+    );
+});
+
+// TODO: Add etherscan test which requires an ZKsync etherscan key.
```

Comment:
Makes sense to add a `ignore = "needs etherscan key"` instead of commenting the test